common/src/main/java/com/wynntils/features/user/ChatMentionFeature.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigHolder;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.handlers.chat.event.ChatMessageReceivedEvent;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.mc.McUtils;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatMentionFeature extends UserFeature {
+    @Config
+    public boolean markMention = true;
+
+    @Config
+    public boolean dingMention = true;
+
+    @Config
+    public String aliases = "";
+
+    private Pattern pattern = Pattern.compile(
+            "(?<!\\[)\\b(" + McUtils.mc().getUser().getName()
+                    + (aliases.length() > 0 ? "|" + aliases.replace(",", "|") : "") + ")\\b(?!:|])",
+            Pattern.CASE_INSENSITIVE);
+
+    @Override
+    protected void onConfigUpdate(ConfigHolder configHolder) {
+        pattern = Pattern.compile(
```

Comment:
This complicated pattern is duplicated twice. That's not a good design.

Please extract the pattern creation to a method, and call it in the constructor and in this method. You can do it either as `updatePattern()` or `pattern = buildPattern()`; whichever you think is the clearest.